### PR TITLE
Login flow update[Breaking Change]

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
This PR adds support for the new API gateway workflow. This is a breaking change and the commits would be squashed and reworded as a Breaking Change before merging.

### CODE REVIEW

**Comment by abhilash-kumar-nair** [comment]
@iakovn , I have now updated this MR with docs and api changes required to migrate to the new login flow. There a few things to consider next:-

- We need to bump the lower bound for the open-api version to ensure, the new version of client fails early against old MI with good error messages.
- Add tags for the old login flow version in RTD
- Update the compatibility matrix

File: docs/source/authentication_with_modelon_impact.rst [hunk 1]
```diff
@@ -24,63 +18,23 @@ in the JupyterHub environment:
     >>>> client = Client()
 
 
-JupyterHub authentication
-*************************
+Authentication outside JupyterHub environment
+*********************************************
```
Comment (iakovn) [inline docs/source/authentication_with_modelon_impact.rst:23]: I suggest merging with the previous section and stop mentioning JupyterHub which is a pure internal component now. We can still be talking "from inside Modelon Impact (either custom functions, scripts or notebooks) vs from outside.

File: docs/source/authentication_with_modelon_impact.rst [hunk 2]
```diff
@@ -4,16 +4,10 @@ Authentication with Modelon Impact
...
-installation, one can create a connection to the external URL. In that case, this will be automatic
-based on the automatically populated ``JUPYTERHUB_API_TOKEN`` environment variable.
-
-An API key must be used in addition to the ``JUPYTERHUB API TOKEN`` to authenticate against the Modelon 
+The cloud/on-premise architecture for Modelon Impact uses an API key to authenticate against the Modelon 
 Impact server operating inside the JupyterHub environment. If the client library is used within the
-JupyterHub environment, this is likewise handled automatically based on the key stored at 
-``~/.impact/api.key`` in that environment.
+JupyterHub environment, this is likewise handled automatically based on the key stored in the environmental
+variable ``MODELON_IMPACT_CLIENT_API_KEY``.
```
Comment (iakovn) [inline docs/source/authentication_with_modelon_impact.rst:10]: Some rewrite and may need to iterate:

In order to work with Modelon Impact REST API a non-browser code needs to pass  `impact-api-key: <api key>` header with all requests. The Python `Client()` constructor will set this header using the value of API key from either (in order of priority):
- explicit constructor parameter or interactively given by a user in interactive mode;
- environment variable `MODELON_IMPACT_CLIENT_API_KEY`;
- `$HOME/.impact/api.key` file.

The `MODELON_IMPACT_CLIENT_API_KEY` environment variable is automatically set with a temporary user specific API key inside the Modelon Impact environment. Therefore explicit user input is not necessary when the Python client is used in a script, notebook or custom function running within the Modelon Impact by a logged in user.

When using Python client from outside Modelon Impact, e.g., from a desktop, the API key needs to be generated via user interface.

File: modelon/impact/client/client.py [hunk 3]
```diff
@@ -240,52 +238,34 @@ def __init__(
...
 
-        self._sal.add_login_retry_with(api_key)
-
-        resp = self._sal.users.get_me()
-        if "license" not in resp["data"]:
-            raise exceptions.NoAssignedLicenseError(
-                "No assigned license during login. Either out of floating Deployment "
-                "Add-on licenses or your assigned user license could not be validated"
-            )
+        # TODO Update to use the unprotected API route https://impact.modelon.cloud/api
```
Comment (iakovn) [inline modelon/impact/client/client.py:267]: Any update on this?

**Review by iakovn — CHANGES_REQUESTED** [review]
Some suggestions for documentation updates.

File: docs/source/authentication_with_modelon_impact.rst (hunk 2 above)
Comment (abhilash-kumar-nair) [reply docs/source/authentication_with_modelon_impact.rst]: Thanks Iakov for the nice write up. I have now made the suggested changes with some minor tweaks to the order of preference. Please let me know if you find this reasonable.

**Comment by iakovn** [comment]
@abhilash-kumar-nair  please review and merge my docs PR. I'm ok with merging to master and integrating after that. Thanks!

**Pull request merged**